ChelseasFieldJournal/src/GUI/GuiDialogToolJournal.cs

```diff
@@ -0,0 +1,202 @@
+            Console.WriteLine("Cooking Entries:" + JournalCookingEntries.Count);
+            Console.WriteLine("Construct Entries:" + JournalConstructEntries.Count);
+
+                // Array of lists containing all your journal entries
+                var journalChapters = new List<object>[] {
+                JournalPageEntries.Cast<object>().ToList(),
+                JournalOreEntries.Cast<object>().ToList(),
+                JournalFaunaEntries.Cast<object>().ToList(),
+                JournalFloraEntries.Cast<object>().ToList(),
+                JournalCookingEntries.Cast<object>().ToList(),
+                JournalConstructEntries.Cast<object>().ToList()
+            };
+
+                if (Chapter >= 0 && Chapter < journalChapters.Length)
+                {
+                    var Entry = journalChapters[Chapter];
+                    if (ChapterLeftPage < Entry.Count)
+                    {
+                        currentLeftPage = Entry[ChapterLeftPage];
+                        currentRightPage = (ChapterLeftPage + 1 < Entry.Count) ? Entry[ChapterLeftPage + 1] : null;
+                    }
+                    else if (Chapter + 1 < journalChapters.Length)
+                    {
+                        Chapter++;
+                        ChapterLeftPage = 0;
+                        currentLeftPage = Entry[ChapterLeftPage];
+                        currentRightPage = (ChapterLeftPage + 1 < Entry.Count) ? Entry[ChapterLeftPage + 1] : null;
+                    }
+                }
+
+            RichTextComponentBase[] richStack = new RichTextComponentBase[] { new ItemstackTextComponent(capi, capi.World.Player.Entity.LeftHandItemSlot.Itemstack, 128.0) };
+
+            base.SingleComposer = gui.CreateCompo(dialogId, dialogBounds).BeginChildElements(childBounds).
+                AddStaticImage(new AssetLocation(MOD_ID + ":" + "textures/dialogs/test.png"), JournalBlankBounds, Operator.Over).
+                AddStaticImage(new AssetLocation(MOD_ID + ":" + "textures/dialogs/1.png"), TestIconBounds, Operator.Atop).
+                AddRichtext(richStack, ItemStackIconBounds).//AddIf(leftPageNumber == 0).
+                AddButton("", new ActionConsumable(this.OnLeftPage), ElementBounds.Fixed(0.0, 0.0, 105.0, 700), EnumButtonStyle.None).
+                AddButton("", new ActionConsumable(this.OnRightPage), ElementBounds.Fixed(986, 0.0, 105.0, 700), EnumButtonStyle.None).
+
+                AddIf(leftPageNumber > 0).
+                AddStaticText(this.leftPageNumber.ToString(), CairoFont.WhiteMediumText().WithWeight(FontWeight.Normal).WithColor(ColorUtil.BlackArgbDouble), ElementBounds.Fixed(60.0, 50.0, 50.0, 50.0), null).
+                EndIf().
+                AddStaticText(this.rightPageNumber.ToString(), CairoFont.WhiteMediumText().WithWeight(FontWeight.Normal).WithColor(ColorUtil.BlackArgbDouble), ElementBounds.Fixed(1000.0, 50.0, 50.0, 50.0), null).
+
```

Comment:
Remove this newline because C# is shit and we can't pipe, so we have to daisy chain dot methods. Keep the comment, though.